### DISCUSSION TITLE
DEFAULT_USER_ROLE broken since 0.3.34

### DISCUSSION THREAD

**Discussion by 0xelion** [open-webui/open-webui#6508] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** 0.3.35
- **Ollama (if applicable):** N/A

- **Operating System:** N/A
- **Browser (if applicable):** N/A

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [NA] I have included the browser console logs.
- [NA] I have included the Docker container logs.
- [NA] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

When setting the `DEFAULT_USER_ROLE` environment variable to `user`. New users should be given the role `user`

## Actual Behavior:

When setting the `DEFAULT_USER_ROLE` environment variable to `user`. New users are given the default role of `pending`

## Description

**Bug Summary:**
the DEFAULT_USER_ROLE environment variable is no longer working since 0.3.34

## Additional Information

New bug that appeared in 0.3.34, users are correctly assigned in 0.3.33

**Comment by tjbck** [comment]
Once your config has been generated it'll ignore your env var, and you'll have to use the UI to configure the default user role.

**Reply by 0xelion** [reply]
> Once your config has been generated it'll ignore your env var, and you'll have to use the UI to configure the default user role.

The problem appeared just after updating to 0.3.34 without changing any configuration. Also the user role is correctly defined in my admin UI.

I forgot to specify but i'm using the OIDC authentication, maybe the problem lies there.

**Reply by 0xelion** [reply]
I was able to solve this problem on my end by using `ENABLE_OAUTH_ROLE_MANAGEMENT` instead and setting up my OIDC provider to give a default role.
